Component consolidation

davidornelas11 opened the pull request:
### Description
This is the main feature branch for the component consolidation effort


### Links

- [JIRA ticket](https://asudev.jira.com/browse/UDS-1571)


### FOR APPROVERS

- [Percy build approval](https://percy.io/5eae92d9/-all-UDS-packages)

davidornelas11 committed:
chore: update storybook preview scss

davidornelas11 committed:
chore(unity-react-core): add individual build files for each component

davidornelas11 committed:
chore(unity-react-core): remove individual UMD component builds

The memory and size reductions were minimal with individual umd component files

davidornelas11 committed:
build(unity-react-core): fix entry files for vite config

davidornelas11 committed:
build(unity-react-core): remove duplicate exports

davidornelas11 committed:
Merge pull request #1363 from ASU/fix-vite-build-indiv

Fix all storybooks, delete webpack files, build individual esm components

davidornelas11 committed:
build: fix storybook build error

scott-williams-az committed:
divider

scott-williams-az committed:
system alerts

scott-williams-az committed:
fix(unity-bootstrap-theme): allow span,i alert-icon tags

scott-williams-az committed:
chore(unity-react-core): autofix and rename stories

scott-williams-az committed:
chore(unity-react-core): switch to vitest

scott-williams-az committed:
chore(unity-react-core): prepare for new ts components

scott-williams-az committed:
chore(unity-react-core): fix some tests

scott-williams-az committed:
update yarn lock

scott-williams-az committed:
lists

scott-williams-az committed:
tsconfig options

scott-williams-az committed:
shared alias

scott-williams-az committed:
tooltip

scott-williams-az committed:
fix shared alias

scott-williams-az committed:
fix some errors ading SB root back in

scott-williams-az committed:
fixing ts errors

scott-williams-az committed:
lint fix

scott-williams-az committed:
Not sure why news started faiing

scott-williams-az committed:
fix alias for tests

scott-williams-az committed:
remove alias

davidornelas11 committed:
feat(component-header-footer): create new component

davidornelas11 committed:
Merge branch 'dev' into rebase-component-consolidation

davidornelas11 committed:
chore: run yarn install

davidornelas11 committed:
chore: upgrade all packages to React 18

davidornelas11 committed:
fix(component-header-footer): sswiotch to styled components insteads of scss file

davidornelas11 committed:
Merge pull request #1377 from ASU/rebase-component-consolidation

Upgrade all components to React 18 and combine header and footer

davidornelas11 committed:
Merge branch 'component-consolidation' into cc-vitest-sbname-plop

davidornelas11 committed:
chore: update tsconfig with proper bundler setting and checkJs files of true

Also added some @ts-nocheck to ignore files we will fix in the future

davidornelas11 committed:
Merge pull request #1370 from ASU/cc-vitest-sbname-plop

Cc vitest sbname plop

davidornelas11 committed:
docs: update examples to use React18

davidornelas11 committed:
test(unity-react-core): add browser tests

N

davidornelas11 committed:
chore(unity-react-core): fix type error

davidornelas11 committed:
test: add playwright to jenkinsfile

davidornelas11 committed:
fix(component-header-footer): update dataLayer pushes to match component-header updates

davidornelas11 committed:
refactor: refactor type hintings

davidornelas11 committed:
test(unity-react-core): update tests

'

mlsamuelson commented on packages/app-degree-pages/public/assets/img/demo-phot-4.jpeg:
Please use a different image. We don't want to include an image in our codebase that has a QR code that we don't know to where it's pointing.

davidornelas11 replied:
Removed

davidornelas11 committed:
chore: remove QR code photo

davidornelas11 committed:
Merge branch 'component-consolidation' into UDS-1714-copy

scott-williams-az committed:
Merge pull request #1353 from ASU/UDS-1714-copy

fix(unity-react-core): updated small ranking card for accessibility

scott-williams-az committed:
update tests

davidornelas11 committed:
Uds 1861-Add security check and remove visual regression testing (#1387)

* build: add security audit for critical and high vulnerabilities

* build: add slack notifictions for security audits

* build: add immutable to yarn install

* build: removed recursive flag

* build: add proper env variaables

* build: fix typo in yarn script

* build: testing nonzero exit of jenmins

* build: remove sh script option

* build: add url

* build: testing prosting to private channel

* build: clean up scripts

* build: only run security audits on PR's

* test: remove percy testing

davidornelas11 committed:
Uds 1861-Add security check and remove visual regression testing (#1387)

* build: add security audit for critical and high vulnerabilities

* build: add slack notifictions for security audits

* build: add immutable to yarn install

* build: removed recursive flag

* build: add proper env variaables

* build: fix typo in yarn script

* build: testing nonzero exit of jenmins

* build: remove sh script option

* build: add url

* build: testing prosting to private channel

* build: clean up scripts

* build: only run security audits on PR's

* test: remove percy testing

davidornelas11 committed:
Uds 1861-Add security check and remove visual regression testing (#1387)

* build: add security audit for critical and high vulnerabilities

* build: add slack notifictions for security audits

* build: add immutable to yarn install

* build: removed recursive flag

* build: add proper env variaables

* build: fix typo in yarn script

* build: testing nonzero exit of jenmins

* build: remove sh script option

* build: add url

* build: testing prosting to private channel

* build: clean up scripts

* build: only run security audits on PR's

* test: remove percy testing

davidornelas11 committed:
Uds 1861-Add security check and remove visual regression testing (#1387)

* build: add security audit for critical and high vulnerabilities

* build: add slack notifictions for security audits

* build: add immutable to yarn install

* build: removed recursive flag

* build: add proper env variaables

* build: fix typo in yarn script

* build: testing nonzero exit of jenmins

* build: remove sh script option

* build: add url

* build: testing prosting to private channel

* build: clean up scripts

* build: only run security audits on PR's

* test: remove percy testing

davidornelas11 committed:
Uds 1861-Add security check and remove visual regression testing (#1387)

* build: add security audit for critical and high vulnerabilities

* build: add slack notifictions for security audits

* build: add immutable to yarn install

* build: removed recursive flag

* build: add proper env variaables

* build: fix typo in yarn script

* build: testing nonzero exit of jenmins

* build: remove sh script option

* build: add url

* build: testing prosting to private channel

* build: clean up scripts

* build: only run security audits on PR's

* test: remove percy testing

davidornelas11 committed:
fix(component-header-footer): add proper keyboard navigation and accessibility for dropdowns

davidornelas11 committed:
fix(component-header-footer): fix build issue

scott-williams-az committed:
fix

scott-williams-az committed:
remove lint on ts

scott-williams-az committed:
remove lint on ts

scott-williams-az committed:
remove lint on ts

scott-williams-az committed:
remove lint on ts

scott-williams-az committed:
remove lint on ts

scott-williams-az committed:
fix

scott-williams-az committed:
fix

davidornelas11 committed:
feat(unity-react-core): added sidebar react implementation

scott-williams-az committed:
wip

scott-williams-az committed:
minor

scott-williams-az committed:
working

scott-williams-az committed:
readme update

davidornelas11 committed:
feat(unity-react-core): add sidebar

mlsamuelson committed:
fix(unity-react-core): enable carousel pinch and zoom

mlsamuelson committed:
chore(unity-react-core): update plop template to use react 18.3.1

mlsamuelson committed:
feat(unity-react-core): add grid links component to react core

mlsamuelson committed:
fix(unity-bootstrap-theme): style tweak to support rendering non-svg icons

scott-williams-az committed:
chore(workspace): fix ts lint

scott-williams-az committed:
chore(workspace): auto fixes

mlsamuelson committed:
Merge pull request #1400 from ASU/cc-ts-lint

Cc ts lint

mlsamuelson committed:
Merge branch 'component-consolidation' into UDS-1869-grid-links

scott-williams-az committed:
chore(static-site): working build

scott-williams-az committed:
chore(static-site): file rename, zIndex

scott-williams-az committed:
Update readme

scott-williams-az committed:
Merge pull request #1395 from ASU/uds-1838-for-consolidation

fix(unity-react-core): enable carousel pinch and zoom

scott-williams-az committed:
Merge pull request #1396 from ASU/plop-template-react18

chore(unity-react-core): update plop template to use react 18.3.1

scott-williams-az committed:
chore(static-site): linting

scott-williams-az committed:
yarn lock

scott-williams-az committed:
Merge branch 'cc' into cc-divider

scott-williams-az committed:
Merge branch 'cc' into cc-lists

scott-williams-az committed:
Merge branch 'cc' into cc-system-alert

scott-williams-az committed:
Merge branch 'cc' into cc-tooltip

scott-williams-az committed:
fix(unity-react-core): fix ts type

mlsamuelson committed:
fix(unity-react-core): fix linting issue with unused variable

mlsamuelson committed:
fix(unity-react-core): add data layer to grid links

scott-williams-az committed:
chore(workspace): remove unused

scott-williams-az committed:
fix lint

mlsamuelson committed:
fix(unity-react-core): improve factoring of data layer render util

mlsamuelson committed:
Merge pull request #1402 from ASU/UDS-1869-datalayer-pattern

fix(unity-react-core): add data layer to grid links

davidornelas11 committed:
style(unity-react-core): formatting code and fix example props

mlsamuelson committed:
fix(unity-react-core): move bgColor to story, out of constants

mlsamuelson committed:
Merge pull request #1397 from ASU/UDS-1869-grid-links

Uds 1869 grid links

mlsamuelson committed:
Merge branch 'component-consolidation' into cc-lists

davidornelas11 committed:
fix(component-header-footer): fix styles in footer

scott-williams-az committed:
chore(unity-react-core): preventing conflict

scott-williams-az committed:
Merge branch 'component-consolidation' into cc-divider

scott-williams-az committed:
Merge branch 'component-consolidation' into cc-system-alert

mlsamuelson committed:
Merge pull request #1388 from ASU/UDS-1865

Uds 1865 - add accessibility in standalone footer

scott-williams-az committed:
feat(unity-react-core): add useExternal to system-alert

scott-williams-az committed:
Update SystemAlert.tsx

scott-williams-az committed:
Update list.html

scott-williams-az committed:
Update divider.html

scott-williams-az committed:
chore(unity-react-core): tooltip/init.js

scott-williams-az committed:
chore(unity-react-core): lists/init.js

scott-williams-az committed:
Update and rename index.js to init.js

scott-williams-az committed:
Rename index.js to init.js

scott-williams-az committed:
chore(unity-react-core): system-alerts test & lint

mlsamuelson committed:
Merge pull request #1381 from ASU/cc-lists

lists

mlsamuelson committed:
Merge branch 'component-consolidation' into cc-tooltip

mlsamuelson committed:
fix(unity-react-core): fix react version in example

mlsamuelson committed:
Merge pull request #1382 from ASU/cc-tooltip

tooltip

mlsamuelson committed:
Merge branch 'component-consolidation' into cc-system-alert

scott-williams-az committed:
feat(unity-react-core): framework switch

mlsamuelson committed:
fix(unity-react-core): add missing required prop for tooltips

mlsamuelson committed:
Merge pull request #1384 from ASU/cc-divider

divider

mlsamuelson committed:
Merge branch 'component-consolidation' into cc-system-alert

mlsamuelson committed:
Merge pull request #1383 from ASU/cc-system-alert

Cc system alert

mlsamuelson committed:
fix(unity-react-core): port uds-1905 fix to consolidation

scott-williams-az committed:
chore(workspace): getting build to work

scott-williams-az committed:
Merge branch 'component-consolidation' of https://github.com/ASU/asu-unity-stack into static-site

scott-williams-az committed:
chore(unity-react-core): removing useExternal

scott-williams-az committed:
chore(workspace): resolving issues

scott-williams-az committed:
feat(unity-react-core): removing useExternal

scott-williams-az committed:
feat(unity-react-core): removing useExternal

scott-williams-az committed:
chore(unity-react-core): more examples GAWrapper

scott-williams-az committed:
chore(workspace): cleanup

scott-williams-az committed:
chore(workspace): making it work

scott-williams-az committed:
Merge pull request #1401 from ASU/UDS-1667b

Uds 1667 - Static Site Generator update as React App

scott-williams-az committed:
Merge branch 'component-consolidation' into cc-framework-switch

scott-williams-az committed:
Merge pull request #1412 from ASU/cc-framework-switch

feat(unity-react-core): framework switch

juanmitriatti committed:
feat(unity-react-core): added functionality to check screen width for modified perview value

scott-williams-az committed:
feat(unity-bootstrap-theme): rank cards native BS

scott-williams-az committed:
feat(unity-react-core): rank cards native BS

Cleanup and use GaEventWrapper

scott-williams-az committed:
feat(unity-react-core): remove all trackGAEvents

All events routed through GaEventWrapper

 .storybook folder
- added functionality to have bootstrap events show in the datalayer panel

scott-williams-az committed:
oopsie

scott-williams-az committed:
chore(data-layer): filter gtm events optional

scott-williams-az committed:
accordion events

scott-williams-az committed:
comment anchor menu

scott-williams-az committed:
comments

davidornelas11 committed:
Merge branch 'component-consolidation' into UDS-1876

scott-williams-az committed:
comments

scott-williams-az committed:
Merge pull request #1417 from ASU/ga-event-wrapper

Ga event wrapper

davidornelas11 committed:
fix(unity-react-core): added basic sidebar story

davidornelas11 committed:
fix(unity-react-core): add Modal story

scott-williams-az committed:
feat(unity-bootstrap-theme): banner icon styles

scott-williams-az committed:
feat(unity-react-core): notification banners

scott-williams-az committed:
feat(unity-react-core): notification banners

non export

scott-williams-az committed:
chore(unity-react-core): update plop

davidornelas11 committed:
chore(unity-react-core): add person profile story

scott-williams-az committed:
fix(unity-bootstrap-theme): apply a11y breadcrumbs min size

davidornelas11 committed:
Merge pull request #1429 from ASU/cc-1917

fix(unity-bootstrap-theme): apply a11y breadcrumbs min size

davidornelas11 committed:
chore(unity-react-core): update Modal

davidornelas11 committed:
Update Modal.tsx

davidornelas11 committed:
chore(unity-react-core): add Table story

davidornelas11 committed:
Merge pull request #1421 from ASU/UDS-1864

fix(unity-react-core): add Modal story

scott-williams-az committed:
fix

scott-williams-az committed:
Merge pull request #1422 from ASU/uds-1866-notification-banner

Uds 1866 notification banner

davidornelas11 committed:
fix(unity-react-core): updated stories for person profile and added GAEventWrapper

davidornelas11 committed:
Merge branch 'component-consolidation' into UDS-1876

davidornelas11 committed:
fix(unity-react-core): removed init file

davidornelas11 committed:
chore(unity-react-core): add image based card story

davidornelas11 committed:
build(unity-bootstrap-theme): update output build filenames

davidornelas11 committed:
docs: update Readme's

davidornelas11 committed:
Merge pull request #1438 from ASU/update-docs

docs: update Readme's

scott-williams-az committed:
Merge pull request #1437 from ASU/uds-1664-js

build(unity-bootstrap-theme): update output build filenames

scott-williams-az committed:
feat(unity-react-core): form elements

davidornelas11 committed:
docs(unity-react-core): remove druplicate story

scott-williams-az committed:
feat(unity-react-core): package dep

scott-williams-az committed:
yarn lock

scott-williams-az committed:
htmlFor

scott-williams-az committed:
expect error

scott-williams-az committed:
feat(unity-react-core): docs

scott-williams-az committed:
fixes

mlsamuelson committed:
Merge pull request #1394 from ASU/UDS-1876

Uds 1876 - Add sidebar

davidornelas11 committed:
docs(unity-react-core): add fixed table story

davidornelas11 committed:
docs(unity-react-core): remove unnecessary gaData spread

davidornelas11 committed:
docs(unity-react-core): remove gaButtonText in interface

mlsamuelson committed:
Merge pull request #1439 from ASU/UDS-1872-forms

feat(unity-react-core): form elements

scott-williams-az committed:
feat(unity-react-core): breadcrumbs

scott-williams-az committed:
fix(unity-bootstrap-theme): breadcrumbs

mlsamuelson committed:
Merge pull request #1427 from ASU/uds-1877

chore(unity-react-core): add person profile story and component

scott-williams-az committed:
Update ImageBasedCard.stories.tsx

mlsamuelson committed:
Merge pull request #1434 from ASU/uds-1871-image-card

chore(unity-react-core): add image based card story

scott-williams-az committed:
Update docPage.tsx

scott-williams-az committed:
Update manager.jsx

scott-williams-az committed:
Clean Up

mlsamuelson committed:
Merge pull request #1442 from ASU/uds-1913

Uds 1913: Storybook Custom doc generator

mlsamuelson reviewed (changes requested):
Issues to resolve before merge:
* [x] unspalsh images need to be replaced in Image, Heroes, Card and wherever they occur.
* [ ] carousel based stories need to force React version. When Bootstrap version displays, they are not interactive.
* [ ] Modal and Tabbed Panels stories throw error on Docs page.
* [x] Static site footer image
* [ ] table fixed story needs to force React version. Bootstrap version is not interactive. Also errors on Docs page.

scott-williams-az committed:
Update Breadcrumbs.tsx

mlsamuelson committed:
Merge pull request #1445 from ASU/uds-1874

feat(unity-react-core): breadcrumbs

mlsamuelson commented:
Also need to ensure that component-header-footer works in WS2 before we merge this. cc @davidornelas11

mlsamuelson commented:
UDS-1912 to update use of unity-react-core internally is still pending, too. cc @scott-williams-az

davidornelas11 committed:
Merge branch 'component-consolidation' into uds-1878

davidornelas11 committed:
docs(unity-react-core): update table story

davidornelas11 committed:
build: updated header-footer and license inclusions

davidornelas11 committed:
fix(component-header-footer): fix header navbar not hiding

mlsamuelson committed:
chore: update yarn.lock

scott-williams-az committed:
Merge pull request #1447 from ASU/update-header-footer

Update header to include fontawesome and fix navbar scrolling issue

mlsamuelson committed:
Merge pull request #1430 from ASU/uds-1878

chore(unity-react-core): add Table story

scott-williams-az committed:
fix(unity-react-core): fix broken images

davidornelas11 committed:
build: add aws s3 staging site

davidornelas11 committed:
build: add commenting on pull requests

davidornelas11 committed:
build: update aws cli image

davidornelas11 committed:
build: add missing bracket

davidornelas11 committed:
build: add new bucket and edit getCurrentScript util

davidornelas11 committed:
build: replace bucket names

mlsamuelson committed:
Merge pull request #1449 from ASU/uds-1930-un-unsplash

fix(unity-react-core): fix broken images

davidornelas11 committed:
fix(component-header-footer): update footer innovation logo

davidornelas11 committed:
build: corrected branch conditions for s3

davidornelas11 committed:
Merge branch 'component-consolidation' into aws-s3-staging

davidornelas11 committed:
build: fix router issue

scott-williams-az committed:
chore(unity-react-core): control render framework per story

davidornelas11 committed:
fix(component-header-footer): update image imports to use vite asset handling

davidornelas11 committed:
build: revert getCurrentScript changes

davidornelas11 committed:
chore(static-site): remove console log

mlsamuelson committed:
Merge pull request #1450 from ASU/aws-s3-staging

Create staging site for each PR

mlsamuelson committed:
Merge pull request #1451 from ASU/UDS-1908

fix(component-header-footer): update footer innovation logo

scott-williams-az committed:
wip

scott-williams-az committed:
fix

mlsamuelson committed:
fix(unity-react-core): fix typo

mlsamuelson commented:
## Root README.md
* https://github.com/ASU/asu-unity-stack/tree/component-consolidation?tab=readme-ov-file#packages-in-this-repository Packages list needs update. Note what's deprecated (move to section below). Add new.

davidornelas11 committed:
feat(app-degree-pages): updated dependecies to new unity-react-core package

Replaced coponents-core with new unity-react-core package

BREAKING CHANGE: Replaced dependency

davidornelas11 committed:
feat(app-webdir-ui): updated to new unity-react-core

BREAKING CHANGE: Replaced components-core with unity-react-core in dependencies

davidornelas11 committed:
feat(app-rfi): updated to new unity-react-core package

Updated to new unity-react-core package

BREAKING CHANGE: Replaced components-core with unity-react-core

scott-williams-az committed:
fix(unity-react-core): fix storybook

davidornelas11 committed:
build(app-degree-pages): switch tests to vitest

davidornelas11 committed:
feat(component-news): replaced components-core with unity-react-core

BREAKING CHANGE: Replaced components-core with unity-react-core

davidornelas11 committed:
feat(component-events): replace components-core with unity-react-core

BREAKING CHANGE: Replace components-core with unity-react-core

davidornelas11 committed:
build(app-degree-pages): removed jest and associated packages

davidornelas11 committed:
build(component-header): removed extra package

davidornelas11 committed:
build(unity-react-core): added type files

Added type files for shared as well to type check across packages

davidornelas11 committed:
docs(unity-bootstrap-theme): updated docs and stories

davidornelas11 committed:
build: adding yarn lock file

davidornelas11 committed:
build(components-core): updated default type for feedcontext

davidornelas11 committed:
build: add unity-react-core in prebuild for package use

davidornelas11 committed:
build: update storybook story config for news and events

mlsamuelson committed:
Merge pull request #1456 from ASU/UDS-1912

Uds 1912 - Updated packages to use unity-react-core

scott-williams-az committed:
resolve merge

davidornelas11 committed:
build(app-rfi): fix storybook build of app-rfi

davidornelas11 committed:
Merge branch 'dev' into component-consolidation

davidornelas11 committed:
docs(unity-bootstrap-theme): update footer story

asu-jenkins-devops commented:
Storybook deployed at https://unity-uds-staging.s3.us-west-2.amazonaws.com/pr-1346/index.html

scott-williams-az committed:
fix

scott-williams-az committed:
fix lockfile

scott-williams-az committed:
fix dep

scott-williams-az committed:
moving stuff

mlsamuelson committed:
Merge pull request #1453 from ASU/uds-1930-framework-control

chore(unity-react-core): control render framework per story

davidornelas11 committed:
docs: update README

davidornelas11 committed:
fix(unity-react-core): update exports in package.json

davidornelas11 committed:
docs(app-degree-pages): updated listing page links for storybook

davidornelas11 committed:
fix(unity-react-core): update package.json

mlsamuelson commented:
Individual PRs reviewed as the went into this feature branch. Also reviewed high level on this branch, and tested S3 static reference site built from this branch.

iodeslykos reviewed (approved):
Approving `Jenkinsfile` and README, because I read it.
